pallets/ethereum-transaction/src/lib.rs

```diff
@@ -0,0 +1,256 @@
+			exit_reason: ExitReason,
+			value: Vec<u8>,
+		},
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Trying to pop from an empty stack.
+		StackUnderflow,
+
+		/// Trying to push into a stack over stack limit.
+		StackOverflow,
+
+		/// Jump destination is invalid.
+		InvalidJump,
+
+		/// An opcode accesses memory region, but the region is invalid.
+		InvalidRange,
+
+		/// Encountered the designated invalid opcode.
+		DesignatedInvalid,
+
+		/// Call stack is too deep (runtime).
+		CallTooDeep,
+
+		/// Create opcode encountered collision (runtime).
+		CreateCollision,
+
+		/// Create init code exceeds limit (runtime).
+		CreateContractLimit,
+
+		/// Invalid opcode during execution or starting byte is 0xef. See [EIP-3541](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-3541.md).
+		InvalidCode(u8),
+
+		/// An opcode accesses external information, but the request is off
+		/// offset limit (runtime).
+		OutOfOffset,
+
+		/// Execution runs out of gas (runtime).
+		OutOfGas,
+
+		/// Not enough fund to start the execution (runtime).
+		OutOfFund,
+
+		/// PC underflowed (unused).
+		PCUnderflow,
+
+		/// Attempt to create an empty account (runtime, unused).
+		CreateEmpty,
+
+		/// The operation is not supported.
+		NotSupported,
+		/// The trap (interrupt) is unhandled.
+		UnhandledInterrupt,
+
+		/// Machine encountered an explicit revert.
+		Reverted,
+
+		/// Unexpected result when executing a transaction.
+		UnexpectedExecuteResult,
+
+		/// Other normal errors.
+		Other,
+	}
+
+	impl<T: Config> Pallet<T> {
+		fn get_transaction_signature() -> Option<TransactionSignature> {
+			//TODO(cdamian): Same signature as the one in ethereum-xcm.
+			TransactionSignature::new(
+				TRANSACTION_RECOVERY_ID,
+				H256::from_low_u64_be(1u64),
+				H256::from_low_u64_be(1u64),
+			)
+		}
+	}
+
+	impl<T: Config> EthereumTransactor for Pallet<T> {
+		/// This implementation serves as a wrapper around the Ethereum pallet
+		/// execute functionality. It keeps track of the nonce used for each
+		/// call and builds a fake signature for executing the provided call.
+		///
+		/// NOTE - The execution fees are charged by the Ethereum pallet,
+		/// we only have to charge for the nonce read operation.
+		fn call(
+			from: H160,
+			to: H160,
+			data: &[u8],
+			value: U256,
+			gas_price: U256,
+			gas_limit: U256,
+		) -> DispatchResultWithPostInfo {
+			let nonce = Nonce::<T>::get();
+			let read_weight = T::DbWeight::get().reads(1);
+
+			let signature =
+				Pallet::<T>::get_transaction_signature().ok_or(DispatchErrorWithPostInfo {
+					post_info: PostDispatchInfo {
+						actual_weight: Some(read_weight),
+						pays_fee: Pays::Yes,
+					},
+					error: DispatchError::Other("Failed to create transaction signature"),
+				})?;
+
+			let transaction = TransactionV2::Legacy(LegacyTransaction {
+				nonce,
+				gas_price,
+				gas_limit,
+				action: TransactionAction::Call(to),
+				value,
+				input: data.into(),
+				signature,
```

Comment:
Using this we can never get a `Create` call right?

Reply:
Yes, that is correct.